opentelemetry-user-events-metrics/src/tracepoint/mod.rs

```diff
@@ -79,5 +79,31 @@ pub unsafe fn register(trace_point: Pin<&ehi::TracepointState>) -> i32 {
     debug_assert!(METRICS_EVENT_DEF[METRICS_EVENT_DEF.len() - 1] == b'\0');
 
     // CStr::from_bytes_with_nul_unchecked is ok because METRICS_EVENT_DEF ends with "\0".
-    trace_point.register(ffi::CStr::from_bytes_with_nul_unchecked(METRICS_EVENT_DEF))
+    // Returns errno code 95 if trace/debug file systems are not mounted
+    // Returns errno code 13 if insufficient permissions
+    // If tracepoint doesn't exist, it will create one automatically
+    let result = panic::catch_unwind(|| {
+        // CStr::from_bytes_with_nul_unchecked is ok because METRICS_EVENT_DEF ends with "\0".
+        trace_point.register(ffi::CStr::from_bytes_with_nul_unchecked(METRICS_EVENT_DEF))
+    });
+
+    match result {
+        Ok(value) => {
+            if value == 0 {
+                0
+            } else {
+                if value == 95 {
+                    eprintln!("Trace/debug file systems are not mounted.");
+                } else if value == 13 {
+                    eprintln!("Insuufficient permissions. You need read/write/execute permissions to user_events tracing directory.");
+                }
+                -1
+            }
+        }
+        // We don't want to ever panic so we catch the error and return a unique code for retry
+        Err(_err) => {
+            eprintln!("Tracepoint failed to register. Try restarting your application.");
```

Comment:
nit - would be good to also print this _err (which seems to be *nix `errrno`) below ?

Reply:
should this use `global::handle_error` for consistency with other otel error logging?